### PR TITLE
Enhance: Make Util::Input detect if mouse button up.

### PR DESCRIPTION
**Pull request by Dragonfly911117**
#81

### CODE REVIEW

File: src/Util/Input.cpp [hunk 1]
```diff
@@ -57,19 +57,37 @@ void Input::Update() {
...
-                       s_Event.button.button == SDL_BUTTON_MIDDLE) ||
-                      s_MBPressed;
-        s_Scroll = s_Event.type == SDL_MOUSEWHEEL || s_Scroll;
+        if (s_LBPressed) {
+            s_LBPressed = s_Event.type != SDL_MOUSEBUTTONUP ||
+                          s_Event.button.button != SDL_BUTTON_LEFT;
+        } else {
+            s_LBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
+                           s_Event.button.button == SDL_BUTTON_LEFT) ||
+                          s_LBPressed;
```
Comment (NOOBDY) [inline src/Util/Input.cpp:69]: isn't the value guaranteed to be `false` in this branch?

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: src/Util/Input.cpp (hunk 1 above)
Comment (Dragonfly911117) [reply src/Util/Input.cpp]: Fixed.

**Review by NOOBDY — DISMISSED** [review]
IWOMM

File: src/Util/Input.cpp [hunk 2]
```diff
@@ -57,19 +57,35 @@ void Input::Update() {
...
-        s_MBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
-                       s_Event.button.button == SDL_BUTTON_MIDDLE) ||
-                      s_MBPressed;
+        if (s_LBPressed) {
+            s_LBPressed = s_Event.type != SDL_MOUSEBUTTONUP ||
+                          s_Event.button.button != SDL_BUTTON_LEFT;
+        } else {
+            s_LBPressed = (s_Event.type == SDL_MOUSEBUTTONDOWN &&
+                           s_Event.button.button == SDL_BUTTON_LEFT);
+        }
```
Comment (ntut-rick) [inline src/Util/Input.cpp:69]: This is making me so confused and triggering my digital logic PTSD
```suggestion
        if (s_Event.type == SDL_MOUSEBUTTONUP
            && s_Event.button.button == SDL_BUTTON_LEFT) {
            s_LBPressed = false;
        }
        if (s_Event.type == SDL_MOUSEBUTTONDOWN
            && s_Event.button.button == SDL_BUTTON_LEFT) {
            s_LBPressed = true;
        }
```

**Review by ntut-rick — CHANGES_REQUESTED** [review]

**Review by ntut-rick — APPROVED** [review]
LGTM

**Pull request merged**